### ISSUE TITLE
maktaba#buffer#Overwrite in neovim sometimes chokes on empty lines

### ISSUE THREAD

**Issue by dbarnett** [google/vim-maktaba#165]
`maktaba#buffer#Overwrite` has a very particular neovim-specific bug causing it to blow up with `NvimError: Index out of bounds` when inserting empty lines in an empty buffer.

To repro, create a fresh buffer and call `maktaba#buffer#Overwrite(1, 1, ['', ''])`:

``` vim
:enew
:call maktaba#buffer#Overwrite(1, 1, ['', ''])
```

```
Error detected while processing function maktaba#buffer#Overwrite..provider#python#Call:
line   18:
NvimError('Index out of bounds',)
Traceback (most recent call last):
  File "/home/dbarnett/.local/lib/python2.7/site-packages/neovim/msgpack_rpc/session.py", line 177, in han
dler
    rv = self._request_cb(name, args)
  File "/home/dbarnett/.local/lib/python2.7/site-packages/neovim/api/common.py", line 237, in filter_reque
st_cb
    walk(self._in, args, self, name, 'request'))
  File "/home/dbarnett/.local/lib/python2.7/site-packages/neovim/plugin/host.py", line 65, in _on_request
    rv = handler(*args)
  File "/usr/local/share/nvim/runtime/autoload/provider/script_host.py", line 76, in python_execute
    exec(script, self.module.__dict__)
  File "<string>", line 1, in <module>
NvimError: Index out of bounds
```

**Comment by malcolmr** [comment]
This might be neovim/neovim#3844, in which case it's recently been fixed.

**Comment by nicdumz** [comment]
I tried a neovim nightly (ee0e214) and I can still reproduce, so I don't think neovim/neovim#3844 is it.

**Comment by nicdumz** [comment]
A minimal test case:

:enew
:py vim.current.buffer[1:1] = ['']

This works in vim, not in neovim

**Comment by nicdumz** [comment]
This issue is now fixed ing neovim nightly, I believe we can close this now

**Comment by dbarnett** [comment]
Fix was neovim/neovim#4083.
